base/base.jl

```diff
@@ -82,7 +82,7 @@ finalize(o::ANY) = ccall(:jl_finalize, Void, (Any,), o)
 gc(full::Bool=true) = ccall(:jl_gc_collect, Void, (Cint,), full)
 gc_enable(on::Bool) = ccall(:jl_gc_enable, Cint, (Cint,), on)!=0
 
-bytestring(str::ByteString) = str
+bytestring(str::String) = str
```

Comment:
Thanks for taking the time to review, @quinnj. Yes, we should deprecate all of `ascii`, `utf8`, `bytestring`, maybe `string` and replace them all with `String`, which will take any number of values, convert each to a string and produce a single `String` object for the concatenation of all of those.